host/sgx/calls.c

```diff
@@ -365,6 +365,47 @@ static oe_result_t _handle_call_host_function(
     return result;
 }
 
+static const char* oe_ocall_str(oe_func_t ocall)
+{
+    switch (ocall)
+    {
+        case OE_OCALL_CALL_HOST_FUNCTION:
+            return "OE_OCALL_CALL_HOST_FUNCTION";
+        case OE_OCALL_GET_QE_TARGET_INFO:
+            return "OE_OCALL_GET_QE_TARGET_INFO";
+        case OE_OCALL_GET_QUOTE:
+            return "OE_OCALL_GET_QUOTE";
+        case OE_OCALL_GET_REVOCATION_INFO:
+            return "OE_OCALL_GET_REVOCATION_INFO";
+        case OE_OCALL_GET_QE_ID_INFO:
+            return "OE_OCALL_GET_QE_ID_INFO";
+        case OE_OCALL_THREAD_WAKE:
+            return "OE_OCALL_THREAD_WAKE";
+        case OE_OCALL_THREAD_WAIT:
+            return "OE_OCALL_THREAD_WAIT";
+        case OE_OCALL_THREAD_WAKE_WAIT:
+            return "OE_OCALL_THREAD_WAKE_WAIT";
+        case OE_OCALL_MALLOC:
+            return "OE_OCALL_MALLOC";
+        case OE_OCALL_REALLOC:
+            return "OE_OCALL_REALLOC";
+        case OE_OCALL_FREE:
+            return "OE_OCALL_FREE";
+        case OE_OCALL_WRITE:
+            return "OE_OCALL_WRITE";
+        case OE_OCALL_SLEEP:
+            return "OE_OCALL_SLEEP";
+        case OE_OCALL_GET_TIME:
+            return "OE_OCALL_GET_TIME";
+        case OE_OCALL_BACKTRACE_SYMBOLS:
+            return "OE_OCALL_BACKTRACE_SYMBOLS";
+        case OE_OCALL_LOG:
+            return "OE_OCALL_LOG";
+        default:
```

Comment:
It depends on the flags we pass. I need to check.

Reply:
@anakrish so the answer is "not currently, no".  We would need to set -Wswitch-enum https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html. But that's irrelevant once this is turned into a string array lookup anyway,

Reply:
@jxyang Can you please investigate if we could add -Wswitch-enum to the set of flags that we compile the OE SDK with? That would be a nice way to get into our cmake build system.

Reply:
I tried that. The thing is, we have many switch statements which don't handle the last enum value (the MAX), for example `__OE_EC_TYPE_MAX`. It will spur quite a few compiler errors if we add this flag.